examples/pytorch/graphsage/link_pred.py

```diff
@@ -68,27 +54,24 @@ def forward(self, pair_graph, neg_pair_graph, blocks, x):
                 h = F.relu(h)
         pos_src, pos_dst = pair_graph.edges()
         neg_src, neg_dst = neg_pair_graph.edges()
-        h_pos = self.predict(h[pos_src], h[pos_dst])
-        h_neg = self.predict(h[neg_src], h[neg_dst])
+        h_pos = self.predictor(h[pos_src] * h[pos_dst])
+        h_neg = self.predictor(h[neg_src] * h[neg_dst])
         return h_pos, h_neg
 
-    def inference(self, g, device, batch_size, num_workers, buffer_device=None):
-        # The difference between this inference function and the one in the official
-        # example is that the intermediate results can also benefit from prefetching.
+    def inference(self, g, device, batch_size):
```

Comment:
Add a comment `"Layer-wise inference algorithm to compute GNN node embeddings."`